openmc/mgxs/mgxs.py

```diff
@@ -1484,22 +1487,26 @@ def rxn_rate_tally(self):
             self._rxn_rate_tally.sparse = self.sparse
         return self._rxn_rate_tally
 
+class FissionXSBase(MGXS):
+    """A fission production multi-group cross section base class
+       for NuFission and KappaFission
+    """
 
-class FissionXS(MGXS):
-    """A fission multi-group cross section."""
+    # This is an abstract class which cannot be instantiated
+    __metaclass__ = abc.ABCMeta
```

Comment:
Good point. It seems like there are two potential options to make the class abstract - add an abstract property and/or method. Perhaps add an abstract property for `rxn_type` in place of the instance attribute `_rxn_type`? Something like this:

```python
class FissionXSBase(MGXS):
...
@abstractproperty
def rxn_type(self):
    return 'fission'

...

class FissionXS(MGXS):
...
@abstractproperty
def rxn_type(self):
    return 'fission'

...

class NuFissionXS(MGXS):
...
@abstractproperty
def rxn_type(self):
    return 'nu-fission'

...

class NuFissionXS(MGXS):
...
@abstractproperty
def rxn_type(self):
    return 'kappa-fission'
```